### PR TITLE
[alertmanager] Unit test sample for ingress version

### LINKED ISSUE #1148: [all] Unit tests
**Is your feature request related to a problem? Please describe.**
I got started with unit tests for charts only recently and would like to suggest them for the project. The scope is simple: test the templates - one file or multiple - as few as possible, as many as necessary, but of course not the entire chart and don't do anything on a k8s cluster.

**Describe the solution you'd like**
I'll provide you with a little preview in #1149 to get you hooked on the stuff, I can't promise any further contribution or that it's not just something you'll close or revert after a few weeks if you don't like the concept.

You need the [helm plugin unittest](https://github.com/quintush/helm-unittest) which is not overboarding with maintainer personal, but it's progressing and working well. And you need to tell your CI to use it. Installation is a one liner.

**Describe alternatives you've considered**
There're value schemes, integration tests, the `helm test` command, but they're not performing tasks that one might see as unit tests. You can't reach any sane test coverage with integration tests - they have their purpose, of course, but it's a different one. Unit tests simply are a different concept that has seen few attention for helm charts. They're part of quality software engineering where everything needs a test - in theory - and in theory there's no difference between theory and practice - you know the drill.

**Additional context**
./.

### PR DESCRIPTION
**Pull request by krichter722**
#### What this PR does / why we need it:
It's an example for unit tests which I encourage to use in #1148. It's an easy demonstration how to quickly cover changes, like supporting a new ingress apiVersion. In this case you'd add

```
  - it: should have apiVersion networking.k8s.io/v1beta1 for k8s >= 1.19
    set:
      ingress.enabled: true
    capabilities:
      majorVersion: 1
      minorVersion: 19
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Ingress
      - isAPIVersion:
          of: networking.k8s.io/v1
```
to the test file which is already covering < 1.14 and >=1.14 and <1.19 in this PR.

Currently the proposed tests are not run on the CI because it needs configuration and thinking and agreement among devs before. Run locally in directory `charts` with `helm unittest --helm3 --strict alertmanager` after `helm plugin install https://github.com/quintush/helm-unittest`.

#### Which issue this PR fixes
  - fixes #1148

#### Special notes for your reviewer:
This is an example for #1148. You probably want to discuss introducing this test mechanism before pressing the merge button to quickly.

#### Checklist
- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: ct.yaml
Comment (krichter722) [inline ct.yaml:16]: In `kube-prometheus-stack` we might want to have directories in the `tests` directory, therefore the second command is setup already. I guess it's easier to do it now rather than figure this out twice.

**Comment by krichter722** [comment]
This now run successfully in the `Run chart-testing (lint)` phase of the `lint-test` githook. And it's blindingly fast.

**Review by naseemkullah — DISMISSED** [review]

**Review by monotek — APPROVED** [review]

**Comment by monotek** [comment]
@torstenwalter @prometheus-community/helm-charts-admins 
please review :)

**Review by torstenwalter — CHANGES_REQUESTED** [review]
@krichter722 Thanks for this it's a big step forward as it enables others to write unittests. Please have a look at my comments/suggestions.

I double checked that also works for charts which do not have unit tests at the moment in https://github.com/prometheus-community/helm-charts/pull/1178/checks?check_run_id=3105175928. Everything is fine :-)

File: .github/workflows/lint-test.yaml
Comment (krichter722) [reply .github/workflows/lint-test.yaml]: It doesn't sign-off the commits, so nevermind. The opposite of amazing.

File: charts/alertmanager/tests/ingress_test.yaml
Comment (torstenwalter) [inline charts/alertmanager/tests/ingress_test.yaml:1]: This file now needs to be moved to `unittests` directory.

File: charts/alertmanager/.helmignore
Comment (torstenwalter) [inline charts/alertmanager/.helmignore:25]:
```suggestion
unittests/
```

**Review by torstenwalter — COMMENTED** [review]
some more adjustments needed to make use of `unittests` directory

**Review by torstenwalter — APPROVED** [review]

**Comment by torstenwalter** [comment]
@monotek for me this looks good. Just needs another review from a chart maintainer.

**Review by monotek — APPROVED** [review]

**Pull request merged**
